docs/testing.md

```diff
@@ -0,0 +1,454 @@
+# Testing with Docket
+
+Docket includes the utilities you need to test all your background task systems in realistic ways. The ergonomic design supports testing complex workflows with minimal setup.
+
+## Testing with Pytest Fixtures
+
+The most powerful way to test with Docket is using pytest fixtures to set up your docket and worker. This approach, used throughout Docket's own test suite, provides clean isolation and reusable test infrastructure.
+
+### Basic Fixture Setup
+
+Create fixtures for your test docket and worker:
+
+```python
+import pytest
+from datetime import datetime, timedelta
+from typing import AsyncGenerator, Callable
+from uuid import uuid4
+from unittest.mock import AsyncMock
+from docket import Docket, Worker
+
+@pytest.fixture
+async def test_docket() -> AsyncGenerator[Docket, None]:
+    """Create a test docket with a unique name for each test."""
+    async with Docket(
+        name=f"test-{uuid4()}",
+        url="redis://localhost:6379/0"
+    ) as docket:
+        yield docket
+
+@pytest.fixture
+async def test_worker(test_docket: Docket) -> AsyncGenerator[Worker, None]:
+    """Create a test worker with fast polling for quick tests."""
+    async with Worker(
+        test_docket,
+        minimum_check_interval=timedelta(milliseconds=5),
+        scheduling_resolution=timedelta(milliseconds=5)
+    ) as worker:
+        yield worker
+
+# Mock task fixture for testing
+@pytest.fixture
+def mock_task() -> AsyncMock:
+    task = AsyncMock()
+    task.__name__ = "mock_task"
+    return task
+
+@pytest.fixture
+def now() -> Callable[[], datetime]:
+    """Consistent time source for tests."""
+    from functools import partial
+    from datetime import timezone
+    return partial(datetime.now, timezone.utc)
+```
+
+### Using Fixtures in Tests
+
+With these fixtures, your tests become much cleaner:
+
+```python
+async def test_task_execution(
+    test_docket: Docket, test_worker: Worker, mock_task: AsyncMock
+):
```

Comment:
We shouldn't be using `mock_task`s in these examples, the user would use their own real tasks!